test/common/deployment_types.go

```diff
@@ -334,8 +407,53 @@ func TestDeploymentConfigExpectedReplicas(t TestingTB, ctx *TestingContext) {
 				)
 				continue
 			}
-
-			if deploymentConfig.Status.AvailableReplicas < product.ExpectedReplicas {
+			if rhmi.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+				if product.Name == "apicast-production" {
+					pods := &corev1.PodList{}
+					selector, _ := labels.Parse("deploymentconfig=apicast-production")
+					listOpts := []k8sclient.ListOption{
+						k8sclient.InNamespace(ThreeScaleProductNamespace),
+						k8sclient.MatchingLabelsSelector{
+							Selector: selector,
+						},
+					}
+					err = ctx.Client.List(context.TODO(), pods, listOpts...)
+					if err != nil {
+						t.Fatalf("failed to get backend listener pods for 3scale: %v", err)
+					}
+					checkDeploymentConfigPods(t, pods, product, namespace, deploymentConfig)
+
```

Comment:
What do you think on having an helper function for getting returning `listOpts` ? Something like
```
func getListOptsForSelectorInNamespace(label, namespace string) []k8sclient.ListOption {
	selector, _ := labels.Parse(label)
	listOpts := []k8sclient.ListOption{
		k8sclient.InNamespace(namespace),
		k8sclient.MatchingLabelsSelector{
			Selector: selector,
		},
	}
	
	return listOpts
}
```
Then the `if` blocks would be smaller in size :thinking: 
```suggestion
					pods := &corev1.PodList{}
					listOpts := getListOptsForSelectorInNamespace("deploymentconfig=apicast-production", ThreeScaleProductNamespace)
					err = ctx.Client.List(context.TODO(), pods, listOpts...)
					if err != nil {
						t.Fatalf("failed to get backend listener pods for 3scale: %v", err)
					}
					checkDeploymentConfigPods(t, pods, product, namespace, deploymentConfig)

```

Reply:
yes, it would make it prettier. Let me try to make it up.